impl/src/main/java/com/stormpath/sdk/impl/oauth/OAuthStormpathTokenGrantAuthenticationAttempt.java

```
@@ -0,0 +1,20 @@
+package com.stormpath.sdk.impl.oauth;
+
+import com.stormpath.sdk.resource.Resource;
+
+/**
+ * @since 1.1.0
+ */
+public interface OAuthStormpathTokenGrantAuthenticationAttempt extends Resource {
+    /**
+     * Method used to set the Authentication Grant Type that will be used for the token exchange request.
+     * @param grantType the Authentication Grant Type that will be used for the token exchange request.
+     */
+    void setGrantType(String grantType);
```

@mraible Yes, totally it will fail, it is a required value. You have to set it internally yourself, I guess in the constructor or always returning a hardcoded value in the get (not sure how it needs to be read). It must not be open in the API. It is confusing and error prone. If you feel you need help I can give it a try. Thanks